src/Site.js

```diff
@@ -539,34 +539,31 @@ class Site {
         logger.warn(e.message);
       }
 
-      // This is to prevent the first nunjuck call from converting {{baseUrl}} to an empty string
-      // and let the baseUrl value be injected later.
-      userDefinedVariables.baseUrl = '{{baseUrl}}';
-      this.userDefinedVariablesMap[base] = userDefinedVariables;
+      /*
```

Comment:
Example of arrow number 1 in the diagram (site variable extraction), similar ones follow in this file